k2/python/k2/fsa_algo.py

```diff
@@ -36,6 +37,33 @@ def linear_fsa(symbols: Union[List[int], List[List[int]]]) -> Fsa:
     return fsa
 
 
+def linear_fst(symbols: Union[List[int], List[List[int]]]) -> Fsa:
```

Comment:
@danpovey 
Can you imagine in which situation is this function helpful?

```python
fsa.aux_labels = fsa.labels.clone()
```
can achieve the same goal in case it is really needed.

Reply:
I agree, there is not really any point to this if the ilabels and olabels are the same, because compose() will treat it the same as a transducer with separate olabels and ilabels.

Reply:
As #554 mentioned, if an FSA does not have aux_labels, compose is equivalent to intersect. So should we modify compose function to treat an acceptor the same as a transducer?

Reply:
I believe it already does.

Reply:
It seems not like that.  
See https://github.com/k2-fsa/k2/blob/bbeb3ea65b1bfe9aec6ba62a06d42a6c78ad19a0/k2/python/k2/fsa_algo.py#L203-L207
I don't thinks this logic will treat an acceptor the same as a transducer.

Reply:
Agree, the above API is meaningless.

I was thinking we were talking about API like below https://github.com/k2-fsa/snowfall/issues/77#issuecomment-764368954:

```
linear_fst(symbols, aux_symbols) where symbols could be a list or a list of list , aux_symbols could be a list or a ragged tensor (2 axes) from single FSA, and be a list of list or a ragged tensor (3 axes).
```
though I prefer approach `fsa.aux_labels = some_tensor/ragged_tensor` for all cases, but I'm fine with above API.

Reply:
OK, this api is more general. I will revise it.

Reply:
sorry we can just use list of list (of list)here instead of ragged tensor, Then the input would be more pythonic